lib/esm-utils.js

```diff
@@ -0,0 +1,35 @@
+// This file is allowed to use async/await because it is not exposed to browsers (see the `eslintrc`),
+// and Node supports async/await in all its non-dead version.
+
+const url = require('url');
+const path = require('path');
+
+exports.requireOrImport = async file => {
+  file = path.resolve(file);
+
+  if (path.extname(file) === '.mjs') {
+    return import(url.pathToFileURL(file));
+  }
+  // This way of figuring out whether a test file is CJS or ESM is currently the only known
+  // way of figuring out whether a file is CJS or ESM.
+  // If Node.js or the community establish a better procedure for that, we can fix this code.
+  // Another option here would be to always use `import()`, as this also supports CJS, but I would be
+  // wary of using it for _all_ existing test files, till ESM is fully stable.
+  try {
+    return require(file);
+  } catch (err) {
+    if (err.code === 'ERR_REQUIRE_ESM') {
+      return import(url.pathToFileURL(file));
```

Comment:
Node doesn't throw ERR_REQUIRE_ESM for .jsx files, instead throws SyntaxError: Cannot use import statement outside a module:
https://github.com/nodejs/node/blob/v13.8.0/lib/internal/modules/cjs/loader.js#L1161

I found a hacky [workaround](https://github.com/mochajs/mocha/pull/4038#issuecomment-587004460).

Could you check `package.json` and import by default if the requiring package has set `type: "module"`?

Reply:
@emma-borhanian this issue is outside of this PR's scope, I would like to get it merged as is.
Afterwards the ESM implementation will grow and we are open to suggestions. But IMO checking for the nearest `package.json` (for each test file) is too expensive and actually Node's job.